api/end_device.proto

```diff
@@ -69,9 +69,11 @@ message MACParameters {
   // ADR: number of retransmissions.
   uint32 adr_nb_trans = 6 [(gogoproto.customname) = "ADRNbTrans", (validate.rules).uint32.lte = 15];
   // ADR: number of messages to wait before setting ADRAckReq.
-  uint32 adr_ack_limit = 7 [(gogoproto.customname) = "ADRAckLimit", (validate.rules).uint32 = {gte: 1, lte: 32768}];
+  // This field is deprecated, use adr_ack_limit_exponent instead.
+  uint32 adr_ack_limit = 7 [(gogoproto.customname) = "ADRAckLimit", deprecated = true];
```

Comment:
Restore the validation rules, also below.

Reply:
The intention is to not store this value in the registry, since it can be derived from the `_exponent` counterpart. Adding validation here breaks flows.